docker_ci_deploy/__main__.py

```diff
@@ -9,25 +9,82 @@
 from itertools import chain
 
 
-def strip_image_tag(image):
+def split_image_tag(image_tag):
     """
-    Remove the tag part from a Docker tag and return the rest. Full tags are of
-    the form [REGISTRYHOST/][NAME/...]NAME[:TAG] where the REGISTRYHOST may
-    contain a ':' but no '/', the NAME parts may contain '/' but no ':', and
-    the TAG part may contain neither ':' nor '/'.
+    Split the given image tag into its name and tag parts (<name>[:<tag>]).
+
+    Full tags are of the form [REGISTRYHOST/][NAME/...]NAME[:TAG] where the
+    REGISTRYHOST may contain a ':' but no '/', the NAME parts may contain '/'
+    but no ':', and the TAG part may contain neither ':' nor '/'.
     """
-    match = re.match('^((?:[^\/]+\/)?[^:]+)(?::[^:\/]+)?$', image)
+    match = re.match(r'^((?:[^\/]+\/)?[^:]+)(?::([^:\/]+))?$', image_tag)
     if match is None:
-        raise RuntimeError('Unable to parse tag "%s"' % (image,))
+        raise ValueError("Unable to parse image tag '%s'" % (image_tag,))
+
+    return match.group(1), match.group(2)
+
+
+def join_image_tag(image, tag):
+    """ Join an image name and tag. """
+    if tag is None or not tag:
```

Comment:
Maybe from Java and friends? :-P
You always want `foo is None` rather than `foo == None` and you generally want to check before calling methods on a thing that could be `None`, but other than that there's no real reason to separate `None` out from falseness if you're checking for falseness anyway and `None` is semantically equivalent.